libs/common/src/platform/services/crypto.service.spec.ts

```diff
@@ -340,4 +348,327 @@ describe("cryptoService", () => {
       });
     });
   });
+
+  describe("userPrivateKey$", () => {
+    const setupKeys = ({
+      makeMasterKey,
+      makeUserKey,
+    }: {
+      makeMasterKey: boolean;
+      makeUserKey: boolean;
+    }): [UserKey, MasterKey] => {
+      const userKeyState = stateProvider.singleUser.getFake(mockUserId, USER_KEY);
+      const fakeMasterKey = makeMasterKey ? makeSymmetricCryptoKey<MasterKey>(64) : null;
+      masterPasswordService.masterKeySubject.next(fakeMasterKey);
+      userKeyState.stateSubject.next([mockUserId, null]);
+      const fakeUserKey = makeUserKey ? makeSymmetricCryptoKey<UserKey>(64) : null;
+      userKeyState.stateSubject.next([mockUserId, fakeUserKey]);
+      return [fakeUserKey, fakeMasterKey];
+    };
```

Comment:
nit: define it as a function, I think that makes it more obvious what it is. Maybe extract the parameter type so it's a bit easier to read

```suggestion
    type SetupKeysParams = { makeMasterKey: boolean; makeUserKey: boolean };
    function setupKeys({ 
        makeMasterKey, 
        makeUserKey 
    }: SetupKeysParams): [UserKey, MasterKey] {
      const userKeyState = stateProvider.singleUser.getFake(mockUserId, USER_KEY);
      const fakeMasterKey = makeMasterKey ? makeSymmetricCryptoKey<MasterKey>(64) : null;
      masterPasswordService.masterKeySubject.next(fakeMasterKey);
      userKeyState.stateSubject.next([mockUserId, null]);
      const fakeUserKey = makeUserKey ? makeSymmetricCryptoKey<UserKey>(64) : null;
      userKeyState.stateSubject.next([mockUserId, fakeUserKey]);
      return [fakeUserKey, fakeMasterKey];
    };
```

Reply:
Implemented in 837f141e1362e01d2a25c979c43c4cbf4a95007c